### PR TITLE
Improvements in the FSI application

### PR DESCRIPTION
**Pull request by rubenzorrilla**
- Creation of a c++ FSI utilities to perform some coupling secondary tasks. 
- Creation of a FSI Python base solver. 
- Displacement based residual.
- Avoid segmentation fault in MVQN strategies when no correction is needed.

### CODE REVIEW

File: applications/FSIapplication/python_scripts/partitioned_fsi_dirichlet_neumann_solver.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+        fluid_interface_residual_size = self.partitioned_fsi_utilities.GetFluidInterfaceResidualSize()
+        self.iteration_value = KratosMultiphysics.Vector(fluid_interface_residual_size)
+        for i in range(0,fluid_interface_residual_size):
+            self.iteration_value[i] = 0.0
+
+
+    def _InitializeDirichletNeumannInterface(self):
+        # Fix the VELOCITY and MESH_VELOCITY variables
+        # MESH_DISPLACEMENT variable is supposed to be already fixed by the set_interface_process
+        for node in self._GetFluidInterfaceSubmodelPart().Nodes:
```
Comment (RiccardoRossi) [inline applications/FSIapplication/python_scripts/partitioned_fsi_dirichlet_neumann_solver.py]: in the variable utilities there is a function to do this in c++

File: applications/FSIapplication/python_scripts/partitioned_fsi_dirichlet_neumann_solver.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+        disp_residual = KratosMultiphysics.Vector(self.partitioned_fsi_utilities.GetFluidInterfaceResidualSize())
+        self.partitioned_fsi_utilities.ComputeFluidInterfaceVectorResidual(KratosALE.MESH_DISPLACEMENT, KratosFSI.VECTOR_PROJECTED, disp_residual)
+
+        return disp_residual
+
+
+    ### INTERFACE MOVEMENT UTILITY ###
+    # Function to update the velocity and acceleration according to the displacement in self.iteration_value.
+    # Note that at the moment only the Bossak scheme is considered
+    def _ComputeCorrectedInterfaceDisplacementDerivatives(self):
```
Comment (RiccardoRossi) [inline applications/FSIapplication/python_scripts/partitioned_fsi_dirichlet_neumann_solver.py:222]: TODO: (for the future) put this in c++ and make it parallel

**Review by RiccardoRossi — APPROVED** [review]
two minor performance comments. for the rest the proposed implementation makes sense to me

File: applications/FSIapplication/python_scripts/partitioned_fsi_dirichlet_neumann_solver.py (hunk 1 above)
Comment (rubenzorrilla) [reply applications/FSIapplication/python_scripts/partitioned_fsi_dirichlet_neumann_solver.py]: Done!

**Pull request merged**